jetty-webapp/src/main/java/org/eclipse/jetty/webapp/StandardDescriptorProcessor.java

```diff
@@ -650,11 +651,10 @@ public void visitSessionConfig(WebAppContext context, Descriptor descriptor, Xml
         XmlParser.Node tNode = node.get("session-timeout");
         if (tNode != null)
         {
-            java.math.BigDecimal asDecimal = new java.math.BigDecimal(tNode.toString(false, true));
-            if (asDecimal.compareTo(org.eclipse.jetty.server.session.SessionHandler.MAX_INACTIVE_MINUTES) > 0)
-                throw new IllegalStateException("Max session-timeout in minutes is " + org.eclipse.jetty.server.session.SessionHandler.MAX_INACTIVE_MINUTES);
-
-            context.getSessionHandler().setMaxInactiveInterval(asDecimal.intValueExact() * 60);
+            int val = Integer.parseInt(tNode.toString(false, true));
+            if (val > SessionHandler.MAX_SESSION_TIMEOUT_MINS)
+                throw new IllegalStateException("Max session-timeout in minutes is " + SessionHandler.MAX_SESSION_TIMEOUT_MINS);
+            context.getSessionHandler().setMaxInactiveIntervalMins(val);
```

Comment:
I would prefer calling a method on `context` here, rather than on `SessionHandler`.
I'm sure people can misconfigure things in a way that `context.getSessionHandler()` return `null`.

Reply:
Well if you don't have a SessionHandler, then it's pointless setting the session timeout. I can check for missing SessionHandler and throw an exception.